setup.py

```diff
@@ -33,7 +33,10 @@
 inc_dirs = [np.get_include()]
 libs = []  # Pre-built libraries ONLY, like python36.so
 clibs = []
-def_macros = []
+def_macros = [
+    # keep in sync with minimal runtime requirement (requirements.txt) 
+    ('NPY_TARGET_VERSION', 'NPY_1_19_API_VERSION')
+]
```

Comment:
AIUI this shouldn't be needed unless numexpr is doing something more particular with the NumPy API

During the conda-forge NumPy 2 bringup discussion we talked to @rgommers about this, Ralf wrote up a nice explanation here ( https://github.com/conda-forge/conda-forge.github.io/issues/1997#issuecomment-2049191501 ). Quoting that below:

> - numpy's `pyproject.toml` contains and enforces the lowest-supported version, e.g.: `requires-python = ">=3.9"`
> - the first numpy version to support that python version (e.g., `1.19.0` for py39) determined the maximum version that `NPY_FEATURE_VERSION` may be set to
> - when that `>=3.9` is bumped to `>=3.10`, then that max-allowed version for `NPY_FEATURE_VERSION` moves up.

So would suggest dropping

```suggestion
def_macros = []
```